packages/node/src/storage/FileSystemStorage.spec.ts

```diff
@@ -20,88 +20,188 @@
-    // ESLint prevents from using new for side-effect, which we want to
-    // do here for test purpose.
-    // eslint-disable-next-line no-new
-    new FileSystemStorage("some file");
-    expect(mockedFileCreation).not.toHaveBeenCalled();
+    it("returns undefined if the key isn't present in storage", async () => {
+      const data = { someKey: "some value" };
+      const mockedFs = jest.requireMock("fs");
+      mockedFs.readFileSync = jest.fn().mockReturnValue(JSON.stringify(data));
+      const storage = new FileSystemStorage("some/path");
+      await expect(storage.get("someOtherKey")).resolves.toBeUndefined();
+    });
   });
 
-  it("does not implement set yet", async () => {
-    const mockedFs = jest.requireMock("fs");
-    mockedFs.existsSync = jest.fn().mockReturnValueOnce(true);
-    const store = new FileSystemStorage("some file");
-    await expect(store.set("someKey", "some value")).rejects.toThrow(
-      "not implemented"
-    );
+  describe("set", () => {
+    it("sets a value for a new key", async () => {
+      const mockedFs = jest.requireMock("fs");
+      mockedFs.existsSync = jest.fn().mockReturnValue(true);
+      mockedFs.readFileSync = jest.fn().mockReturnValue("{}");
+      mockedFs.promises = {
+        writeFile: jest.fn(),
+      };
+      const storage = new FileSystemStorage("some/path");
+      await storage.set("someKey", "some value");
+      await expect(storage.get("someKey")).resolves.toEqual("some value");
+    });
+
+    it("overwrites an existing value", async () => {
+      const data = { someKey: "some old value" };
+      const mockedFs = jest.requireMock("fs");
+      mockedFs.existsSync = jest.fn().mockReturnValue(true);
+      mockedFs.readFileSync = jest.fn().mockReturnValue(JSON.stringify(data));
+      mockedFs.promises = {
+        writeFile: jest.fn(),
+      };
+      const storage = new FileSystemStorage("some/path");
+      await storage.set("someKey", "some new value");
+      await expect(storage.get("someKey")).resolves.toEqual("some new value");
+    });
+
+    it("persists the new value on disk", async () => {
+      const mockedFs = jest.requireMock("fs");
+      mockedFs.existsSync = jest.fn().mockReturnValue(true);
+      mockedFs.readFileSync = jest.fn().mockReturnValue("{}");
+      mockedFs.promises = {
+        writeFile: jest.fn(),
+      };
+      const storage = new FileSystemStorage("some/path");
+      await storage.set("someKey", "some value");
+      expect(mockedFs.promises.writeFile).toHaveBeenCalledWith(
+        "some/path",
+        JSON.stringify({ someKey: "some value" })
+      );
+    });
+
+    it("throws in case of write error", async () => {
+      const mockedFs = jest.requireMock("fs");
+      mockedFs.existsSync = jest.fn().mockReturnValue(true);
+      mockedFs.readFileSync = jest.fn().mockReturnValue("{}");
+      mockedFs.promises = {
+        writeFile: () => {
+          throw new Error("Some write error");
```

Comment:
Given that it's an async function, I think this is a more accurate simulation of its behaviour:

```suggestion
          return Promise.reject(new Error("Some write error"));
```